### DISCUSSION TITLE
How to create an empty branch on a GitHub repo?

### DISCUSSION THREAD

**Discussion by toshikijp** [andatoshiki/toshiki-discuss#19] [Q&A] [answered]
As the question describes.

**Comment by andatoshiki** [comment, marked as answer]
November 2021 Update: As of git version 2.27, you can now use git switch --orphan <new branch> to create an empty branch with no history.

Unlike `git checkout --orphan <new branch>`, this branch won't have any files from your current branch (save for those which git doesn't track).

This should be the preferred way to create empty branches with no prior history.

Once you actually have commits on this branch, it can be pushed to github via git push -u origin <branch name>:

```sh
git switch --orphan <new branch>
git commit --allow-empty -m "Initial commit on orphan branch"
git push -u origin <new branch>
```

Reference: [@stackoverflow](https://stackoverflow.com/questions/34100048/create-empty-branch-on-github)

see #7
